File: BasicClassValidation/Rules/ListHasElements.vb
Imports System.ComponentModel.DataAnnotations
Imports BasicClassValidation.HelperModules
Namespace Rules
    Public Class ListHasElements
        Inherits ValidationAttribute

        Public Overrides Function IsValid(sender As Object) As Boolean

            If sender Is Nothing Then
                Return False
            End If

            If IsList(sender) Then
                Dim result = CType(sender, IEnumerable).Cast(Of Object)().ToList()
                Return result.Any()
            Else
                Return False
            End If

        End Function
    End Class
End Namespace

File: BaseDataValidatorLibrary/CommonRules/ListHasElements.cs
using System.Collections;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using BaseDataValidatorLibrary.LanguageExtensions;

namespace BaseDataValidatorLibrary.CommonRules
{
    public class ListHasElements : ValidationAttribute
    {
        public override bool IsValid(object sender)
        {
            if (sender == null)
            {
                return false;
            }

            if (sender.IsList())
            {
                var result = ((IEnumerable)sender).Cast<object>().ToList();
                return result.Any();
            }
            else
            {
                return false;
            }
        }
    }
}
